File: app/src/main/java/com/tendebit/dungeonmaster/charactercreation/classselection/model/CharacterClassDirectory.java
package com.tendebit.dungeonmaster.charactercreation.classselection.model;

import com.google.gson.annotations.SerializedName;
import com.tendebit.dungeonmaster.core.SelectionElement;

import org.jetbrains.annotations.NotNull;

public class CharacterClassDirectory implements SelectionElement {
    @SerializedName("name")
    public String name;
    @SerializedName("url")
    public String url;

    @Override
    public String toString() {
        return "Entry for class " + name + " can be found at " + url;
    }

    @NotNull
    @Override
    public String primaryText() {
        return name;
    }
}

File: app/src/main/java/com/tendebit/dungeonmaster/charactercreation/pages/classselection/model/CharacterClassDirectory.kt
package com.tendebit.dungeonmaster.charactercreation.pages.classselection.model

import com.google.gson.annotations.SerializedName
import com.tendebit.dungeonmaster.core.viewmodel.SelectableElement

/**
 * Model object corresponding to the basic class info sent by the dnd5e API
 */
class CharacterClassDirectory : SelectableElement {
    @SerializedName("name")
    lateinit var name: String
    @SerializedName("url")
    lateinit var url: String

    override fun toString(): String {
        return "Entry for class $name can be found at $url"
    }

    override fun primaryText(): String {
        return name
    }

    override fun primaryId(): String {
        return url
    }
}